pkg/utils/imagefilterlist.go

```diff
@@ -0,0 +1,142 @@
+package utils
+
+import (
+	"errors"
+	"github.com/gobwas/glob"
+	"log"
+	"regexp"
+	"strings"
+)
+
+const defaultContainerRegistry = "docker.io"
+
+// ImageFilterList represents a list of glob filters that can be used to check if an image matches one of these filters
+type ImageFilterList struct {
+	registryMatcher *regexp.Regexp
+	userMatcher     *regexp.Regexp
+	patterns        []glob.Glob
+}
+
+// NewAllowAllImageFilterList creates a new empty list, which will allow all images
+func NewAllowAllImageFilterList() (*ImageFilterList, error) {
+	return NewImageFilterList([]string{})
+}
+
+// BuildImageAllowList creates a ImageFilterList from a comma separated string that is present as environment variable
+func BuildImageAllowList(envVariable string) (*ImageFilterList, error) {
+	// Extract allow list from env variable, strip empty strings from the
+	// list, since they are useless, and we really don't want them
+	var allowListStrings []string
+	for _, str := range strings.Split(envVariable, ",") {
+		if str != "" {
+			allowListStrings = append(allowListStrings, str)
+		}
+	}
+
+	// Remind the user that he is probably running an unsafe configuration
+	if len(allowListStrings) == 0 {
+		log.Println("Found empty allowlist for images, all images are allowed!")
+	}
+
+	return NewImageFilterList(allowListStrings)
+}
+
+// NewImageFilterList creates a new list of wildcards
+func NewImageFilterList(patterns []string) (*ImageFilterList, error) {
+
+	registryMatcher, err := regexp.Compile("^.*\\..*/")
+	if err != nil {
+		return nil, err
+	}
+
+	userMatcher, err := regexp.Compile("^.*\\.*/.*/")
+	if err != nil {
+		return nil, err
+	}
+
+	// transform all patterns to a compiled regex instance
+	globPatterns := make([]glob.Glob, len(patterns))
+	for index, pattern := range patterns {
+
+		// If the pattern is a * we can skip everything and just create an empty list
+		if pattern == "*" {
```

Comment:
Maybe add a warning here since this could be a simple typo that would allow any image (think about an allowlist like `docker.io/*, ghcr.io,*` which doesn't make much sense...)

We could even be stricter and consider such allow list as invalid and only allow stuff like `*` but nothing else...